openid-4-verifiable-presentations-1_0.md

```diff
@@ -1596,15 +1624,14 @@ try {
 Out of the Authorization Request parameters defined in [@!RFC6749] and (#vp_token_request), the following are supported with this profile: 
 
 * `client_id`
-* `client_id_scheme`
 * `response_type`
 * `response_mode`
 * `nonce`
 * `presentation_definition`
 * `client_metadata`
 * `request`
 
-The `client_id` and `client_id_scheme` MUST be omitted in unsigned requests defined in (#unsigned_request). The Wallet determines the Client Identifier from the origin as asserted by the Web Platform and/or app platform. The transport of the request and origin from the Web Platform and/or app platform to the Wallet is platform-specific and is out of scope of this profile.
+The `client_id` MUST be omitted in unsigned requests defined in (#unsigned_request). The Wallet determines the Client Identifier from the origin as asserted by the Web Platform and/or app platform. The transport of the request and origin from the Web Platform and/or app platform to the Wallet is platform-specific and is out of scope of this profile.
```

Comment:
Per my (possibly flawed) understanding, this effectively means that an unsigned request will have a Client Identifier determined to be something like "https://verifiers.domain.com" which is a value that "indicates that the Client Identifier is an Entity Identifier defined in OpenID Federation" https://github.com/openid/OpenID4VP/pull/263/files#diff-3118c9756a1d7b361bb53af8bd9d65666476a27cfeeced56c47a5dccc38eac55R494 

It is not that. 

This bit of text was arguably a bit unspecified before this PR. 

But now it's wrong.

Reply:
As discussed on the call, an option would be that the wallet adds a prefix indicating that it derived the client id from the web platform, e.g., `api-orign:`.

Reply:
This option was more or less the agreed to path forward in the call (as I remember it anyway).

Reply:
I still object to special treatment of DIDs and Federation. And do not buy the argument about DIDs not being special. But in the interest of moving forward, I've conceded on that point and agreed on fixing things here. And also came to realize that #213 is related.

Reply:
Here is an attempt at some suggested text, which I half promised to do during the call.

```suggestion
The `client_id` parameter MUST be omitted in unsigned requests defined in (#unsigned_request). The Wallet determines the effective Client Identifier from the origin as asserted by the Web Platform and/or app platform. The effective Client Identifier is composed of a synthetic Client Identifier Scheme of `web-orign` and the origin itself. For example, an origin of `https://verifier.example.com` would result in an effective Client Identifier of `web-orign:https://verifier.example.com` The transport of the request and origin from the Web Platform and/or app platform to the Wallet is platform-specific and is out of scope of this profile.
```

The text in the {#unsigned_request} section likely needs a bit of an update too. But that content is too far out of scope from the changes in the PR to make a suggestion.

Reply:
WRT the text in {#unsigned_request} perhaps,

> The Verifier MAY send all the OpenID4VP request parameters as members in the `request` member passed to the API. In this case, the Wallet will use the Verifier's origin as asserted by the user agent as the Verifer's Client Identifier. 

to

> The Verifier MAY send all the OpenID4VP request parameters as members in the `request` member passed to the API. In this case, the Wallet will use the Verifier's origin as asserted by the user agent to determine the Verifier's effective Client Identifier.

or something

Reply:
Thank you Dr. @danielfett